Src/AutoNSubstituteUnitTest/SubstituteAttributeRelayTest.cs

```diff
@@ -0,0 +1,163 @@
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteAttributeRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNull()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(null, context);
+            // Verify outcome
+            Assert.IsType<NoSpecimen>(specimen);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotICustomAttributeProvider()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = new object();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(specimen);
+            Assert.Same(request, noSpecimen.Request);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenICustomAttributeProviderDoesNotReturnExpectedAttributeType()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            request.GetCustomAttributes(Arg.Any<Type>(), Arg.Any<bool>()).Returns(new object[0]);
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object specimen = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(specimen);
+            Assert.Same(request, noSpecimen.Request);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateGetsInheritedAttributesFromProviderBecauseAttributesAreInheritedByDefault()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ICustomAttributeProvider>();
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            sut.Create(request, context);
+            // Verify outcome
+            request.Received().GetCustomAttributes(typeof(SubstituteAttribute), true);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsSpecimenResolvedByContextFromSubstituteRequest()
+        {
+            // Fixture setup
+            var sut = new SubstituteAttributeRelay();
+            var request = Substitute.For<ParameterInfo>();
+            request.ParameterType.Returns(typeof(IInterface));
+            request.GetCustomAttributes(Arg.Any<Type>(), Arg.Any<bool>()).Returns(new[] { new SubstituteAttribute() });

```

Comment:
What if you changed this line to the following?

```C#
request
    .GetCustomAttributes(typeof(SubstituteAttribute), Arg.Any<bool>())
    .Returns(new[] { new SubstituteAttribute() });
```

Wouldn't the `CreateGetsInheritedAttributesFromProviderBecauseAttributesAreInheritedByDefault` test be redundant, then?